resip/stack/SipMessage.hxx

```diff
@@ -737,6 +751,8 @@ class SipMessage : public TransactionMessage
 
       std::vector<MessageDecorator*> mOutboundDecorators;
 
+      SipMessageOptions mOpts;
```

Comment:
The SipMessage class is one we need to be really careful about to not add unnecessary bloat to.  It largely affects the memory use of any SIP server (as there can be 1000's or them allocated). If SipMessageOptions grows to be larger, and it is not used by most people, we'd probably be better off implementing it as dynamically allocated data, so that "extra" space only get's used if needed.  That being said, it only contains 2 int's now, which is the same size as a 64bit pointer, ...but I'm still thinking ahead to when it might get expanded for more uses.

Reply:
I see.

> but I'm still thinking ahead to when it might get expanded for more uses.

Would you expect both `SipMessageOptions` and `TxOptions` to be stored in a dynamic memory?

Reply:
I was thinking SipMessageOptions as a whole, hadn't really considered more fine grained....